packages/kas/src/nodes.ts

```diff
@@ -3744,42 +3809,36 @@ var siPrefixes = {
 };
 
 // Use these two values to mark a unit as either SI-prefixable or not.
-var hasPrefixes = {};
-var hasntPrefixes = {};
+const hasPrefixes = Symbol();
+const hasntPrefixes = Symbol();
+
+type Prefixes = typeof hasPrefixes | typeof hasntPrefixes;
 
-var makeAlias = function (str, prefixes) {
+const makeAlias = function (str: string, prefixes: Prefixes) {
     var splits = str.split("|");
     var coefficientStr = splits[0].trim();
     var unitsStr = splits[1].trim();
 
-    var coefficient = Num.One;
+    var coefficient = NumOne;
     if (coefficientStr !== "") {
         coefficient = parse(coefficientStr).expr;
     }
 
     var numdenomStr = unitsStr.split("/");
-    var numdenom = [coefficient];
+    var numdenom: Expr[] = [coefficient];
 
     if (numdenomStr[0]) {
         numdenomStr[0]
             .split(" ")
-            .filter(function (x) {
-                return x !== "";
-            })
-            .map(function (x) {
-                numdenom.push(new Unit(x));
-            });
+            .filter((x) => x !== "")
+            .map((x) => numdenom.push(new Unit(x)));
```

Comment:
**nit against the original code:** I think using `map` here instead of `forEach` is semantically incorrect.

Reply:
I didn't even notice that.  Good eye.  There really should be a lint rule for useless `.map()` calls.  I ran into the same thing in webapp in a couple of places as well.